administrator/components/com_workflow/resources/scripts/app/WorkflowGraphApi.es6.js

```diff
@@ -0,0 +1,183 @@
+import Event from './Event.es6.js';
+
+class WorkflowGraphApi {
+  constructor() {
+    const options = Joomla.getOptions('com_workflow', {});
+    if (!options.apiBaseUrl) {
+      throw new TypeError('Workflow API baseUrl is not defined');
+    }
+    this.baseUrl = options.apiBaseUrl;
+
+    if (!options.extension) {
+      throw new TypeError('Workflow API extension is not defined');
+    }
+    this.extension = options.extension;
+    this.csrfToken = Joomla.getOptions('csrf.token');
+    if (!this.csrfToken) {
+      console.warn('CSRF token not found');
+    }
+  }
+
+  /**
+   * Generic request method with better error handling
+   */
+  async makeRequest(url, options = {}) {
```

Comment:
Probably worth checking out: https://github.com/joomla/joomla-cms/blob/6.0-dev/build/media_source/system/js/core.es6.js#L600(and use it)